src/monodroid/jni/internal-pinvoke-api.cc

```diff
@@ -0,0 +1,326 @@
+#include <unistd.h>
+#include <stdarg.h>
+#include <mono/utils/mono-publib.h>
+
+#include "xa-internal-api.hh"
+
+constexpr int TRUE = 1;
+constexpr int FALSE = 0;
+
+using namespace xamarin::android;
+
+static MonoAndroidInternalCalls *internal_calls = nullptr;
+
+MONO_API bool
+_monodroid_init_internal_api (MonoAndroidInternalCalls *api)
+{
+	if (api == nullptr)
+		return false;
+
+	internal_calls = api;
+	return true;
+}
+
+MONO_API int
+_monodroid_getifaddrs (struct _monodroid_ifaddrs **ifap)
+{
+	return internal_calls->monodroid_getifaddrs (ifap);
+}
+
+MONO_API void
+_monodroid_freeifaddrs (struct _monodroid_ifaddrs *ifa)
+{
+	internal_calls->monodroid_freeifaddrs (ifa);
+}
+
+MONO_API void
+_monodroid_detect_cpu_and_architecture (unsigned short *built_for_cpu, unsigned short *running_on_cpu, unsigned char *is64bit)
+{
+	internal_calls->monodroid_detect_cpu_and_architecture (built_for_cpu, running_on_cpu, is64bit);
+}
+
+/* !DO NOT REMOVE! Used by Mono BCL (System.Net.NetworkInformation.NetworkInterface) */
```

Comment:
As per previous discussions, I would like to start having "full" references to what's using the code, so that if/when the referencing code is updated, we can more easily determine that a function is removable, e.g.

https://github.com/mono/mono/blob/e59c1cd70f4a7171a0ff5e1f9f4937985d6a4d8d/mcs/class/System/System.Net.NetworkInformation/LinuxNetworkInterface.cs#L250-L261